redis.go

```diff
@@ -1,46 +1,184 @@
+//
+// The type of the underlying client depends on the following conditions:
+// 1. If the MasterName option is specified, a sentinel-backed FailoverClient is used.
+// 2. if the number of Addrs is two or more, a ClusterClient is used.
+// 3. Otherwise, a single-node Client is used.
+func (mi *ModuleInstance) NewClient(call goja.ConstructorCall) *goja.Object {
+	rt := mi.vu.Runtime()
+
+	var optionsArg map[string]interface{}
+	err := rt.ExportTo(call.Arguments[0], &optionsArg)
+	if err != nil {
+		common.Throw(rt, errors.New("unable to parse options object"))
+	}
+
+	opts, err := newOptionsFrom(optionsArg)
+	if err != nil {
+		common.Throw(rt, fmt.Errorf("invalid options; reason: %w", err))
+	}
+
+	client := &Client{
+		vu: mi.vu,
+		redisClient: redis.NewUniversalClient(&redis.UniversalOptions{
+			Addrs:              opts.Addrs,
+			DB:                 opts.DB,
+			Username:           opts.Username,
+			Password:           opts.Password,
+			SentinelUsername:   opts.SentinelUsername,
+			SentinelPassword:   opts.SentinelPassword,
+			MasterName:         opts.MasterName,
+			MaxRetries:         opts.MaxRetries,
+			MinRetryBackoff:    opts.MinRetryBackoff,
+			MaxRetryBackoff:    opts.MaxRetryBackoff,
+			DialTimeout:        opts.DialTimeout,
+			ReadTimeout:        opts.ReadTimeout,
+			WriteTimeout:       opts.WriteTimeout,
+			PoolSize:           opts.PoolSize,
+			MinIdleConns:       opts.MinIdleConns,
+			MaxConnAge:         opts.MaxConnAge,
+			PoolTimeout:        opts.PoolTimeout,
+			IdleTimeout:        opts.IdleTimeout,
+			IdleCheckFrequency: opts.IdleCheckFrequency,
+			MaxRedirects:       opts.MaxRedirects,
+			ReadOnly:           opts.ReadOnly,
+			RouteByLatency:     opts.RouteByLatency,
+			RouteRandomly:      opts.RouteRandomly,
+		}),
+	}
+
+	return rt.ToValue(client).ToObject(rt)
 }
 
-// Set the given key with the given value and expiration time.
-func (c *Client) Set(key, value string, exp time.Duration) {
-	c.client.Set(c.client.Context(), key, value, exp)
+type options struct {
+	// Either a single address or a seed list of host:port addresses
+	// of cluster/sentinel nodes.
+	Addrs []string
+
+	// Database to be selected after connecting to the server.
+	// Only used in single-node and failover modes.
+	DB int
+
+	// Use the specified Username to authenticate the current connection
+	// with one of the connections defined in the ACL list when connecting
+	// to a Redis 6.0 instance, or greater, that is using the Redis ACL system.
+	Username string
+
+	// Optional password. Must match the password specified in the
+	// requirepass server configuration option (if connecting to a Redis 5.0 instance, or lower),
+	// or the User Password when connecting to a Redis 6.0 instance, or greater,
+	// that is using the Redis ACL system.
+	Password string
+
+	SentinelUsername string
+	SentinelPassword string
+
+	MasterName string
+
+	MaxRetries      int
+	MinRetryBackoff time.Duration
+	MaxRetryBackoff time.Duration
+
```

Comment:
This will either parse `"5m30s"` or be in nanoseconds instead of milliseconds as is the case everywhere else across k6 (we talked about this in other places). 

I guess for consistency it will be nice to have this work the same, but in this case this will mean (probably a lot of work .. or a custom type :thinking: